manifests/repo/apt.pp

```diff
@@ -1,19 +1,24 @@
 # PRIVATE CLASS: do not use directly
-class mongodb::repo::apt inherits mongodb::repo {
+class mongodb::repo::apt(
+  $location = $mongodb::repo::location,
+  $release  = $mongodb::repo::release,
+  $repos    = $mongodb::repo::repos,
+  $key      = {
```

Comment:
I'd prefer to refactor `mongodb::repo` to provide `$key` as a hash instead. It'd keep it very straight forward.